torch/autograd/gradcheck.py

```diff
@@ -349,6 +350,41 @@ def fn(input):
             if gi.size() != i.size():
                 return fail_test('grad is incorrect size')
 
+        if check_undefined_grad:
+            def check_undefined_grad_support(output_to_check):
+                grads_output = [torch.zeros_like(o, memory_format=torch.legacy_contiguous_format) for o in output_to_check]
+                try:
+                    grads_input = torch.autograd.grad(output_to_check,
+                                                      diff_input_list,
+                                                      grads_output,
+                                                      allow_unused=True)
+                except RuntimeError:
+                    raise RuntimeError((
+                        'Expected backward function to handle undefined output grads. '
+                        'Please look at "Notes about undefined output gradients" in '
+                        '"tools/autograd/derivatives.yaml"'))
```

Comment:
I think you want to provide the original error in this message. To help the user debug where the error comes from exactly.

Reply:
I had that same thought, but when I exercise this error through test_nn.py, I also get the original error.

```
Traceback (most recent call last):
  ...
  File "/work/kurtamohler/development/pytorch-undefined-tensor-gradients/torch/autograd/__init__.py", line 190, in grad
    inputs, allow_unused)
RuntimeError: Expected a Tensor of type Variable but found an undefined Tensor for argument #0 'grad_out'
Exception raised from checked_cast_variable at
...


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  ...
  File "/work/kurtamohler/development/pytorch-undefined-tensor-gradients/torch/autograd/gradcheck.py", line 363, in check_undefined_grad_support
    'Expected backward function to handle undefined output grads. '
RuntimeError: Expected backward function to handle undefined output grads. Please look at "Notes about undefined output gradients" in "tools/autograd/derivatives.yaml"

```
But I'm not 100% familiar with Python exceptions, maybe the `During handling of the above exception` part isn't guaranteed to be included?

Reply:
Looks like python 3.0+ always prints both exceptions: https://www.python.org/dev/peps/pep-3134/